routing and login/sign up pages

affan880 opened the pull request:
- Added routing capabilities using react-router-dom with new Login an…

- Refactored App component to utilize Routes for navigation.

affan880 committed:
- Added routing capabilities using react-router-dom with new Login and Register pages.
- Refactored App component to utilize Routes for navigation.

jackli921 commented on apps/client/src/lib/schema.tsx:
```diff
@@ -0,0 +1,67 @@
+      .max(30, { message: 'Password must be less than 30 characters' })
+      .regex(
+        /[^a-zA-Z0-9]/,
+        'Password must contain at least one special character',
```
is this a repeat of line 25? Looks like the requirement is repeated and this one doesn't didn't wrap the message in an object?

affan880 replied:
yes, It was a mistake. Fixing it

jackli921 commented on apps/client/src/lib/schema.tsx:
```diff
@@ -0,0 +1,67 @@
+    .max(30, { message: 'Password must be less than 30 characters' })
+    .regex(
+      /[^a-zA-Z0-9]/,
+      'Password must contain at least one special character',
```
is this the same regex requirement as line 62? Remove this one and keep the one below?

affan880 replied:
yes, fixing it

jackli921 replied:
Looks like removed the LoginSchema entirely? I think you'd still need it for the resolver in the useForm() in the LoginPage right? Right now the useForm function doesn't have a resolver for validating the input.

Edit: Ok. I just saw Michael comment below on the not giving away detailed validation errors for security best practices. Yeah, we'll just leave the validation on the backend then.

jackli921 replied:
I think we should still have a basic schema to check to prevent user submitting login request with empty username or password fields. Right now you can click the login button with both field empty.

jackli921 commented on apps/client/src/pages/Auth/LoginPage.tsx:
```diff
@@ -0,0 +1,151 @@
+                  </label>
+                </div>
+                <a
+                  href="/forgot-password"
```
There is no backend logic for retrieving/resetting password in plan for mvp at least, is this input going to be a placeholder?

affan880 replied:
Got it - we’ll remove the password reset input entirely for now.

jackli921 commented on apps/client/src/pages/Auth/LoginPage.tsx:
```diff
@@ -0,0 +1,151 @@
+
+              <div className="flex items-center justify-between mt-2">
+                <div className="flex items-center">
+                  <input
```
how do we want to move forward with this functionality? what would this functionality entail?

kayoMichael replied:
This is theoretically possible to be implemented by using refresh tokens to refresh the JWT whenever it expires but I don't know if its needed in the MVP.

affan880 replied:
Thanks for the Input Jack, Michael. You're right Michael. It is possible to implement remember me but as of now I think we should consider removing it.

jackli921 replied:
Feel free to add the idea to the Optional Features channel so that we can revisit and add it later! I'll resolve this comment for now.

jackli921 commented on apps/client/src/pages/Auth/LoginPage.tsx:
```diff
@@ -0,0 +1,151 @@
+              onSubmit={handleSubmit(onSubmit)}
+              className="space-y-4 md:space-y-6"
+            >
+              <div className="flex flex-col gap-1.5 md:gap-2">
```
How did you decide on the styling of these parts?

No issue with the implementation of the react-hook-forms as I think this is the standard of of implementing it.
I've played around with using the forms component from shad-cn and it uses react-hook-form under the hood.

affan880 replied:
sorry, I didn't get it. styling of overall page?

jackli921 replied:
The choices of spacing, gaps, color, etc. 
Since there is no such form in the demo app, I was curious how you decided on the overall look of the forms.

affan880 replied:
for the previous one.

I took the Inspiration from [Here](https://accessible-astro-dashboard.netlify.app/login/)

jackli921 replied:
I see. Looks geat.

jackli921 commented on apps/client/src/pages/Auth/LoginPage.tsx:
```diff
@@ -0,0 +1,151 @@
+      </div>
+
+      {/* Image Section */}
+      <div className="md:block hidden w-2/3 h-screen bg-white">
```
Nice idea, but I don't think having the piggy image is necessary for two reasons:
1. alignment - currently the input form is pushed to the left side on a full sized 30-inch screen, which looks weird. It should be just cenetered.
2. styling - the piggy bank image conveys a totally differnet design language compared to the landing page and the rest of the app. An empty background or background with the same color as the demo app look more consistent.

For reference:
![image](https://github.com/user-attachments/assets/8fb6571f-8961-4c77-9471-2f9cb9869186)

affan880 replied:
I thought It was a good idea. since, the page was looking very empty

jackli921 replied:
I see. White space is not necessarily a bad thing. In fact, it's often used to provide emphasize and breathing room for what's important, which is the form itself. The new changes with the form centered really spotlights the form well I think.

jackli921 reviewed (changes requested):
Looks good overall. I think you could add the navigate function to the other two sign up buttons on the landing page. Currently, only the nav buttons redirects.

kayoMichael commented on apps/client/src/lib/schema.tsx:
```diff
@@ -0,0 +1,67 @@
+    path: ['confirmPassword'],
+  });
+
+export const loginSchema = z.object({
```
I don't think we should have these validations in login if we are following security best practices. Since these give clues to what the password or username might be. In a standard login flow, the validation should be the server checking the password, and the client giving an ambiguous error message like "Username or password is incorrect."

affan880 replied:
Good Point. Noted, fixing it

affan880 committed:
feat: enhance authentication pages with improved UI and functionality

- Added password visibility toggle for both Login and Register pages.
- Updated button states to reflect loading status during form submission.
- Refactored form handling to improve user experience and error handling.
- Integrated navigation to the registration page from the landing components.

jackli921 commented on apps/client/src/pages/Auth/LoginPage.tsx:
```diff
@@ -0,0 +1,137 @@
+                type="submit"
+                disabled={isLoading}
+                className="w-full py-2.5 md:py-3 mt-2 md:mt-4
+                bg-indigo-600 hover:bg-indigo-700 
```
is this the default color from react-hook-forms or shad? What was the reason for the indigo color? 

Asking because the last time we discussed the overall look of the app, we were going to use shadcn's default look, but if we aren't doing that, I think we can use the demo app's UI color choice as a fall back. E.g., the buttons on the app have a dark/black background.

![image](https://github.com/user-attachments/assets/09539d2b-3249-486b-a585-2492fe064e87)

affan880 replied:
There is no reason for it. I just put it there since I wanted to have some color to the button.


Since we haven't decided the color schema. It is just a default one.

jackli921 replied:
Let's strive for consistency. We can keep the purple color as it is, but we'll just apply it across the app, or we could fall back to the demo app's color choice. Which one do you all prefer? @kayoMichael  @affan880  @Shaimaa01

jackli921 commented:
Not sure where to put it but, I think we should have a link somewhere on both forms to redirect user back to the homepage. Right now there is no visible navigation to revisit the landing back once the user lands on either login or register page. It doesn't need to be a call to action button like Login/Register as that's the main CTA button for the page, but something like a colored text? What do you guys think? @kayoMichael @affan880 @Shaimaa01

jackli921 reviewed (changes requested):
See newest comment in the discussion. That one is not a hard request for change but just a suggestion.

affan880 committed:
Enhance navigation and UI in authentication pages

- Added a back navigation button with a ChevronLeft icon to both Login and Register pages.
- Updated button styles in the landing navigation to include cursor pointer for better user interaction.

affan880 committed:
Merge branch 'development' into login/signup-page
